### DISCUSSION TITLE
Label support all characters

### DISCUSSION THREAD

**Discussion by ComputerDev** [axmolengine/axmol#598] [Q&A]
Hello I would like to know what would be the best approach for a label to support all characters also support outline and set a color for each letters ?

Would RichText be what I'm looking for ? Wouldn't it be too expensive to segment text to create color, put emoticons... Whenever I see an impossibility like an emoticons for a TTF or a new color I create a new RichTextElement to meet the need.

![Sans titre](https://user-images.githubusercontent.com/91702377/155976111-411ed4f2-9421-4018-b427-36529b1ce032.png)

	auto rich = ui::RichText::create();
	rich->setPosition(Vec2(100, 200));
	this->addChild(rich);

	auto text = ui::RichElementText::create(0, Color3B::BLUE, 255, "Hello 😂", "fonts/troika.ttf", 20);
	rich->insertElement(text, 0);

	rich->insertElement(ui::RichElementNewLine::create(1, Color3B::RED, 255), 1);

	text = ui::RichElementText::create(2, Color3B::GREEN, 255, "Hello ", "fonts/troika.ttf", 20, ui::RichElementText::UNDERLINE_FLAG | ui::RichElementText::URL_FLAG, "https://www.google.com/");
	rich->insertElement(text, 2);

	text = ui::RichElementText::create(3, Color3B::GREEN, 255, "😂", "Segoe UI Semibold", 20);
	rich->insertElement(text, 3);

	text = ui::RichElementText::create(4, Color3B::GREEN, 255, "😚", "Segoe UI Semibold", 20);
	rich->insertElement(text, 4);

	auto label = Label::createWithTTF("Hello 😂", "fonts/troika.ttf", 20);
	label->setPosition(Vec2(100, 100));
	this->addChild(label);

**Comment by aismann** [comment]
@ComputerDev ,
have you found a solution?
Maybe your result is usefull?
Or can it be deleted?

**Reply by ComputerDev** [reply]
No solution found
For languages I use specific font

**Comment by aismann** [comment]
Have you check also the cpp-test examples?
![image](https://user-images.githubusercontent.com/8652787/194481118-78e99614-912a-4907-9f60-32c05e833739.png)
![image](https://user-images.githubusercontent.com/8652787/194481349-d83588bd-38ca-4fa4-bd14-1e5c11e54af6.png)
![image](https://user-images.githubusercontent.com/8652787/194481549-f109cb88-4db5-420a-8b02-9472dff8b637.png)
![image](https://user-images.githubusercontent.com/8652787/194481649-9c22a72c-abfb-4192-be89-e2c08da36be9.png)
![image](https://user-images.githubusercontent.com/8652787/194481808-847fc84f-6830-43b0-8b27-7e8d7fc12adf.png)
![image](https://user-images.githubusercontent.com/8652787/194482359-c00fea8e-f497-4513-bd0e-f9b80865ad23.png)

**Comment by ComputerDev** [comment]
I mean label with native emoticons

**Reply by DelinWorks** [reply]
You're gonna need to use RichText Node to accomplish this and add custom emoticons to that richText node through a texture.

**Reply by ComputerDev** [reply]
Thank you
